libs/core/langchain_core/messages/ai.py

```diff
@@ -196,6 +197,79 @@ def lc_attributes(self) -> dict:
             "invalid_tool_calls": self.invalid_tool_calls,
         }
 
+    @property
+    def beta_content(self) -> list[types.ContentBlock]:
+        """Return the content as a list of standard ContentBlocks.
+
+        To use this property, the corresponding chat model must support
+        ``output_version="v1"`` or higher:
+
+        .. code-block:: python
+
+            from langchain.chat_models import init_chat_model
+
+            llm = init_chat_model("...", output_version="v1")
+
+        otherwise, does best-effort parsing to standard types.
+        """
+        blocks: list[types.ContentBlock] = []
+        if isinstance(self.content, str):
+            if self.content:
+                blocks.append({"type": "text", "text": self.content})
+            else:
+                pass
+
+        elif isinstance(self.content, list):
+            for item in self.content:
+                if isinstance(item, str):
+                    blocks.append({"type": "text", "text": item})
+
+                elif isinstance(item, dict):
+                    item_type = item.get("type")
+                    if item_type == "text":
+                        blocks.append(cast("types.TextContentBlock", item))
```

Comment:
The casts aren't needed here since they don't affect anything on the user's end.

The user only sees ```def beta_content(self) -> list[types.ContentBlock]:``` this type signature.

The logic here only needs to reshape the actual data to match the types.ContentBLock convention